cocos/renderer/CCGLProgram.cpp

```diff
@@ -553,17 +547,11 @@ bool GLProgram::link()
     parseVertexAttribs();
     parseUniforms();
 
-    if (_vertShader)
-    {
-        glDeleteShader(_vertShader);
-    }
-
-    if (_fragShader)
-    {
-        glDeleteShader(_fragShader);
-    }
-
-    _vertShader = _fragShader = 0;
+#if CC_TARGET_PLATFORM != CC_PLATFORM_MAC && CC_TARGET_PLATFORM != CC_PLATFORM_WIN32
+    // Keep _vertShader & _fragShader on PC for cocos studio, because them 
+    //  are needed for 3d object click detection in 3d scene editing.
+    clearShader();
+#endif
```

Comment:
This logic is not correct. 
If Cocos Studio needs a feature, then it should say either: `#if COCOS_STUDIO...` or add a methods that prevents clearing the shaders like `void dontClearShaders()`.